ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -60,6 +101,23 @@ const SnapUIRendererComponent = ({
     [content],
   );
 
+  const footerSections = useMemo(
```

Comment:
I was hoping we didn't need to map content and footer individually. Is there a reason this is preferred to mapping over everything once?

Reply:
The confirmation's page CSS is tricky to handle (it's a grid layout and we would need to disable it to get the layout good) and we need to guard against people using `Footer` outside of the custom dialog. There's nothing that prevents it other than that

Reply:
It's not obvious to me how mapping it individually helps with this. The result looks very similar to what I would expect would happen if we map all of it at the same time.

How does the current implementation guard against using Footer outside of a dialog?